homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,269 @@
+    """Set up the Group Cover platform."""
+    async_add_devices(
+        [GroupCover(config.get(CONF_NAME), config.get(CONF_ENTITIES))])
+
+
+class GroupCover(CoverDevice):
+    """Representation of a GroupCover."""
+
+    def __init__(self, name, entities):
+        """Initialize a GroupCover entity."""
+        self._name = name
+        self._tilt = False
+
+        self.entities = entities
+        self.covers = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                       KEY_POSITION: set()}
+        self.tilts = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                      KEY_POSITION: set()}
+
+    def update_supported_features(self, entity_id, old_state, new_state):
+        """Update dictionaries with supported features."""
+        if new_state is None:
+            for value in self.covers.values():
+                value.discard(entity_id)
+            for value in self.tilts.values():
+                value.discard(entity_id)
+                tilt = True if value else False
+            self._tilt = tilt
+            return
+
+        if old_state is None:
+            features = new_state.attributes[ATTR_SUPPORTED_FEATURES]
+            if features & (SUPPORT_OPEN | SUPPORT_CLOSE):
+                self.covers[KEY_OPEN_CLOSE].add(entity_id)
+            if features & (SUPPORT_STOP):
+                self.covers[KEY_STOP].add(entity_id)
+            if features & (SUPPORT_SET_POSITION):
+                self.covers[KEY_POSITION].add(entity_id)
+            if features & (SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT):
+                self.tilts[KEY_OPEN_CLOSE].add(entity_id)
+                self._tilt = True
+            if features & (SUPPORT_STOP_TILT):
+                self.tilts[KEY_STOP].add(entity_id)
+                self._tilt = True
+            if features & (SUPPORT_SET_TILT_POSITION):
+                self.tilts[KEY_POSITION].add(entity_id)
+                self._tilt = True
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def state_change_listener(entity, old_state, new_state):
+            """Handle cover state changes and update state."""
+            self.update_supported_features(entity, old_state, new_state)
+            self.async_schedule_update_ha_state(True)
+
+        for entity_id in self.entities:
+            new_state = self.hass.states.get(entity_id)
+            self.update_supported_features(entity_id, None, new_state)
+        async_track_state_change(self.hass, self.entities,
+                                 state_change_listener)
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def assumed_state(self):
+        """Enable buttons even if at end position."""
+        return True
+
+    @property
+    def should_poll(self):
+        """Disable polling for the MultiCover."""
+        return False
+
+    @property
+    def is_closed(self):
```

Comment:
I'm not sure, but I think the properties can in theory be called quite often, therefore it would be more efficient to store the `is_closed`, `current_cover_position`, ... as attributes like `self._is_closed` and save them during something like `async_update`.